### PR TITLE
fix(zimage): align SGLD CFG combine formula with diffusers Z-Image

### PR DESCRIPTION
**Pull request by Prozac614**
## Motivation

SGLD's Z-Image output diverges from HuggingFace diffusers' `ZImagePipeline` reference: on Z-Image-Turbo + LoRA (8 steps, guidance=3.5) SSIM is only 0.139 and PSNR 9.52 dB, far below the visual-match threshold. Two independent issues are responsible:

1. Diffusers Z-Image uses a non-standard CFG combine `pred = -(pos + scale * (pos - neg))`, while SGLD's `_combine_cfg_serial` uses the standard `pred = uncond + scale * (cond - uncond)`. Combined with the Z-Image DiT's internally inverted output sign (`return -torch.stack(x)`), the two formulas differ by exactly one unit of `scale`.
2. `ZImageTurboSamplingParams.negative_prompt` defaulted to `None`. `Req.validate()` only flips `do_classifier_free_guidance=True` when the negative prompt is non-`None` and `guidance_scale > 1.0`, so CFG was silently disabled on Turbo even when callers raised `guidance_scale`. Non-Turbo `ZImageSamplingParams` already uses `" "` for this reason.

## Modifications

- `python/sglang/multimodal_gen/configs/pipeline_configs/zimage.py`: override `ZImagePipelineConfig.get_classifier_free_guidance_scale` to return `guidance_scale + 1.0`, making SGLD's standard combine algebraically equivalent to diffusers' formula given the DiT's sign inversion.
- `python/sglang/multimodal_gen/configs/sample/zimage.py`: change `ZImageTurboSamplingParams.negative_prompt` default from `None` to `" "` so CFG engages whenever `guidance_scale > 1.0`. The single space satisfies `V.string_not_none` in `TextEncodingStage`.

Default `ZImageTurboSamplingParams.guidance_scale` stays `0.0`, so unmodified default callers (e.g. AMD nightly Z-Image-Turbo test) take no CFG path and are unaffected by either change.

## Accuracy Tests

Tongyi-MAI/Z-Image-Turbo + anime LoRA, 8 steps, guidance=3.5, vs diffusers reference:

| metric           | before | after |
| ---------------- | ------ | ----- |
| SSIM             | 0.139  | 0.908 |
| PSNR (dB)        | 9.52   | 20.38 |
| final-latent cos | 0.607  | 0.977 |

**~0.2% residual remains in the final latents because SGLD's optimized Qwen3 text encoder (fused kernels + custom attention) differs numerically from diffusers' transformers.Qwen3Model under bf16, and the gap compounds through the 30-layer DiT. Closing it requires swapping the text encoder, which is a separate precision/performance trade-off out of scope for this PR.**

Tongyi-MAI/Z-Image (non-Turbo), 50 steps, guidance=5.0: no regression (SSIM 0.824 → 0.864).

comparison_3panel

comparison_3panel

comparison_3panel

## Speed Tests and Profiling

No speed impact. The change is a constant-time scalar adjustment in `get_classifier_free_guidance_scale` and a default-value change in `SamplingParams`; the model forward, scheduler and CFG engine are unchanged.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

### CODE REVIEW

File: python/sglang/multimodal_gen/configs/pipeline_configs/zimage.py [hunk 1]
```diff
@@ -177,6 +177,9 @@ def _get_zimage_sp_plan(self, batch) -> dict:
             plan = self._build_zimage_sp_plan(batch)
         return plan
 
+    def get_classifier_free_guidance_scale(self, batch, guidance_scale: float) -> float:
+        return guidance_scale + 1.0
```
[the remark this answers is not in the document]
Comment (Prozac614) [reply python/sglang/multimodal_gen/configs/pipeline_configs/zimage.py:184]: Fixed

**Review by mickqian — APPROVED** [review]
should we update the consistency check?

**Comment by Prozac614** [comment]
> should we update the consistency check?

It seems this bug case isn’t covered in CI yet—I’ll add a test case for it. After that, we’ll need to regenerate the GT.
